### PR TITLE
Added consumes calls to ES modules in RecoLocalMuon/RPCRecHit

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Modified the ESProducers in the package to make consumes calls. As part of that moved to using ESGetTokens.

In addition, removed Record get calls from ES data products and moved
them into their respective ES modules.

#### PR validation:

The code compiles. 
The changes are technical and should not change results.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: RecoLocalMuon/RPCRecHit/src/CSCObjectMap.cc [hunk 1]
```diff
@@ -9,14 +9,8 @@
 #include "RecoLocalMuon/RPCRecHit/src/CSCObjectMap.h"
 #include "RecoLocalMuon/RPCRecHit/src/CSCStationIndex.h"
 
-CSCObjectMap::CSCObjectMap(MuonGeometryRecord const& record) {
-  edm::ESHandle<RPCGeometry> rpcGeo;
-  record.get(rpcGeo);
-
-  edm::ESHandle<CSCGeometry> cscGeo;
```
Comment (perrotta) [inline RecoLocalMuon/RPCRecHit/src/CSCObjectMap.cc]: If you remove this (useless, indeed) you can also remove the include of "Geometry/CSCGeometry/interface/CSCGeometry.h"

File: RecoLocalMuon/RPCRecHit/src/DTObjectMap.cc [hunk 2]
```diff
@@ -10,14 +10,8 @@
 #include "RecoLocalMuon/RPCRecHit/src/DTObjectMap.h"
 #include "RecoLocalMuon/RPCRecHit/src/DTStationIndex.h"
 
-DTObjectMap::DTObjectMap(MuonGeometryRecord const& record) {
-  edm::ESHandle<RPCGeometry> rpcGeo;
-  record.get(rpcGeo);
-
-  edm::ESHandle<DTGeometry> dtGeo;
```
Comment (perrotta) [inline RecoLocalMuon/RPCRecHit/src/DTObjectMap.cc]: If you remove this (useless, indeed) you can also remove the include of "Geometry/DTGeometry/interface/DTGeometry.h"

File: RecoLocalMuon/RPCRecHit/src/CSCObjectMap.cc (hunk 1 above)
Comment (Dr15Jones) [reply RecoLocalMuon/RPCRecHit/src/CSCObjectMap.cc]: done

File: RecoLocalMuon/RPCRecHit/src/DTObjectMap.cc (hunk 2 above)
Comment (Dr15Jones) [reply RecoLocalMuon/RPCRecHit/src/DTObjectMap.cc]: done

**Comment by Dr15Jones** [comment]
code-checks

**Comment by Dr15Jones** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by kpedro88** [comment]
+1

**Comment by makortel** [comment]
part of #26748

**Pull request merged**